mne/encoding/model.py

```diff
@@ -0,0 +1,325 @@
+            two parameters: X and y, and return two parameters corresponding to
+            X and y after the function has been applied.
+
+        References
+        ----------
+        .. [1] Theunissen, F. E. et al. Estimating spatio-temporal receptive
+               fields of auditory and visual neurons from their responses to
+               natural stimuli. Network 12, 289-316 (2001).
+        .. [2] Willmore, B. & Smyth, D. Methods for first-order kernel
+               estimation: simple-cell receptive fields from responses to
+               natural scenes. Network 14, 553-77 (2003).
+        """
+        self.est = _check_estimator(est)
+        self.preproc_y = _check_preproc(preproc_y)
+        self.preproc_x = _check_preproc(preproc_x)
+        self.preproc_func_xy = preproc_func_xy
+
+    def fit(self, X, y=None):
+        """Fit the model.
+
+        Fits a receptive field model. Model results are stored as attributes.
+
+        Parameters
+        ----------
+        X : array, shape (n_times, n_features)
+            The data on which we want to fit a regression model.
+        y : array, shape (n_times, n_channels)
+
+        Attributes
+        ----------
+        coefs_ : array, shape (n_channels, n_features)
+            The coefficients fit on each row of X
+        coef_names : array, shape (n_features,)
+            A list of coefficient names, useful for keeping track of time lags
+        """
+        if y is None:
+            raise ValueError('Must supply an output variable `y`')
+        if X.shape[0] != y.shape[0]:
+            raise ValueError('X and y must have last dimension same length.')
+        n_times, n_chs = y.shape
+
+        # Prepare the input feature matrix
+        if self.preproc_x is not None:
+            X = self.preproc_x.fit_transform(X)
+        if self.preproc_y is not None:
+            y = self.preproc_y.fit_transform(y)
+
+        if self.preproc_func_xy is not None:
+            X, y = self.preproc_func_xy(X, y)
+
+        # Fit the model and assign coefficients
+        self.est.fit(X, y)
+        self.X_ = X
+        self.y_ = y
+        return self
+
+    def predict(self, X, preproc_x=None):
+        """Generate predictions using fit coefficients.
+
+        This uses the predict method of the final estimator in the
+        `est` attribute.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input feature array.
+        """
+        # Preprocess X with the pipeline
+        if preproc_x is not None:
+            X = preproc_x.fit_transform(X)
+        return self.est.predict(X)
+
+
+def _check_estimator(est):
+    from sklearn.linear_model import Ridge
+    from sklearn.pipeline import Pipeline
+    if est is None:
+        est = Ridge()
+    elif isinstance(est, string_types):
+        if est not in estimator_dict.keys():
+            raise ValueError("No such solver: {0}".format(est))
```

Comment:
OK - so if we're switching to a sklearn `Ridge` instance with alpha == 0, I could allow for any of the strings that the `Ridge` solver would allow for. We could say in the docs that passing a string will be passed to `Ridge(solver=...)` with alpha == 0.